lib/game_state.rb

```diff
@@ -32,6 +45,19 @@ def previous_player
 
   private
 
+  def validate_move(position)
```

Comment:
does it have to return two booleans? one could be enough maybe?

Reply:
I had two different errors, so I returned two booleans, but I agree that it would be more ideal to return one.

I changed my code to return one boolean. One tradeoff is that I have to run `@move_validator.valid_integer?` twice. I think this is acceptable considering that the method is very lightweight, and it practically doesn't get any slower when increasing board size (unlike `@move_validator.empty_square?`).

Reply:
or maybe the message about invalid move can be called somewhere else and you will not have to call `@move_validator.valid_integer?` twice?

btw, the loop inside `make_move` looks so much better :)

Reply:
I can't think of a good way to call only call `@move_validator.valid_integer?` once. 

Maybe I could save the results of the calls to instance variables, but I would think side effects aren't great. Did you have something specific in mind?